File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> frequencies;
        for(int i=0; i<nums.size(); i++){
            frequencies[nums[i]]++;
        }
        priority_queue<pair<int,int>> pq;
        for(auto& p: frequencies){
            pq.push({p.second, p.first});
        }
        vector<int> result;
        for(int i=0; i<k; i++){
            result.push_back(pq.top().second);
            pq.pop();
        }
        return result;
    }
};

File: Data Structures & Algorithms/asteroid-collision/submission-1.py
class Solution:
    def asteroidCollision(self, asteroids: List[int]) -> List[int]:
        s = []
        for n in asteroids:
            while s and n<0 and s[-1]>0:
                diff = s[-1] + n
                if diff>0:
                    n=0
                elif diff <0:
                    s.pop()
                else:
                    n=0
                    s.pop()
            if n:
                s.append(n)
        return s



        